sys/usb/usbus/cdc/acm/cdc_acm.c

```diff
@@ -126,7 +126,26 @@ static size_t _gen_full_acm_descriptor(usbus_t *usbus, void *arg)
 /* Submit (ACM interface in) */
 size_t usbus_cdc_acm_submit(usbus_cdcacm_device_t *cdcacm, const uint8_t *buf, size_t len)
 {
-    return tsrb_add(&cdcacm->tsrb, buf, len);
+    size_t n;
+    if (cdcacm->state != USBUS_CDC_ACM_LINE_STATE_DISCONNECTED) {
+        irq_disable();
+        n = tsrb_add(&cdcacm->tsrb, buf, len);
+        irq_enable();
```

Comment:
I'm aware that it is confusing, but if I'm correct this should be:
```
unsigned old = irq_disable();
n = tsrb_add(&cdcacm->tsrb, buf, len); 
irq_restore(old);
```

Same for the other cases

Reply:
Thanks. Fixed.